force_wfmanager/left_side_pane/variable_names_registry.py

```diff
@@ -50,50 +57,68 @@ def __init__(self, workflow, *args, **kwargs):
 
     @on_trait_change(
         'workflow.mco.parameters.name,'
-        'workflow.execution_layers.data_sources.output_slot_info.name')
+        'workflow.execution_layers.data_sources.output_slot_info.name,'
+        'workflow.mco.parameters.type,'
+    )
     def update_available_variables_stack(self):
         stack = []
 
         # At the first layer, the available variables are the MCO parameters
         if self.workflow.mco is None:
             stack.append([])
         else:
-            stack.append([
-                p.name
-                for p in self.workflow.mco.parameters
-                if len(p.name) != 0
-            ])
+            stack.append([{'name': p.name, 'type': p.type}
```

Comment:
I am not sure about this. I assumed you had a mapping from identifier to type, that is p.name: p.type.